include/solvers/twostep_time_solver.h

```diff
@@ -71,13 +71,27 @@ class TwostepTimeSolver : public AdaptiveTimeSolver
 
   virtual std::pair<unsigned int, Real> adjoint_solve (const QoISet & qoi_indices) override;
 
+  /**
+   * A method to integrate the system::QoI functionals.
+   */
+  virtual void integrate_qoi_timestep() override;
+
   /**
    * A method to integrate the adjoint sensitivity w.r.t a given parameter
    * vector. int_{tstep_start}^{tstep_end} dQ/dp dt = int_{tstep_start}^{tstep_end} (\partialQ / \partial p) - ( \partial R (u,z) / \partial p ) dt
    * The midpoint rule is used to integrate each substep
    */
   virtual void integrate_adjoint_sensitivity(const QoISet & qois, const ParameterVector & parameter_vector, SensitivityData & sensitivities) override;
 
+  /**
+   * A method to compute the adjoint refinement error estimate at the current timestep.
+   * int_{tstep_start}^{tstep_end} R(u^h,z) dt
+   * Fills in an ErrorVector that contains the weighted sum of errors from ARefEE::qoi_set() and can be used to guide AMR.
+   * Also fills in a map that links QoI indices to spatially integrated error estimates for the QoI with that index.
+   * The midpoint rule is used for the numerical integration.
```

Comment:
This is no longer correct, though, right?  We're now summing the substeps integrals, and using whatever our nested time solvers use, which in general might not be the midpoint rule?